### DISCUSSION TITLE
Has anyone used javaagent on apache spark?

### DISCUSSION THREAD

**Discussion by imperio-wxm** [open-telemetry/opentelemetry-java-instrumentation#8019] [Q&A]
I am trying to connect opentelemetry to apache spark, but how the context and span created by driver transmit it to executors. 

The idea is to broadcast traceid and spanid to each executor via broadcast variables, and then use propagation to extract traceid+spanid as context to continue using on the executor.

Is there a best practice for this?

**Comment by imperio-wxm** [comment]
Has anyone answered this question?

**Reply by trask** [reply]
hi @imperio-wxm, there is no auto-instrumentation for Apache Spark currently. you could try manual context propagation

**Comment by patpatpat123** [comment]
Hello team, two years after the original post, with Spark 4 and java agent 2, it is still not working.

Would it be possible to get some help on this initiative?

This would help each and every project avoid manual code to manage the context propagation. 

Thank you

**Reply by Neutrinic** [reply]
Hi, I ran into this exact problem and built an open-source library to solve it: https://github.com/Neutrinic/flare

It handles W3C traceparent propagation from driver to executors automatically via a ByteBuddy agent extension, no broadcast variables or manual context propagation needed. The result is a full app → sql → job → stage → task span hierarchy across both JVMs. Spark 4.0 is supported alongside 3.3–3.5.
Hope it helps.
